src/types.rs

```diff
@@ -2074,9 +2115,38 @@ impl RuleContext {
         let rhs_str = self.rhs.iter().map(|rhs| rhs.pretty(sig)).join(" | ");
         format!("{} = {}", lhs_str, rhs_str)
     }
-    /// Get all the subcontexts and [`Place`]s in a `RuleContext`.
+    /// Get all the [`subcontexts`] and [`Place`]s in a `RuleContext`.
     ///
+    /// [`Subcontexts`]: struct.Context.html
     /// [`Place`]: type.Place.html
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use term_rewriting::{Signature, Term, RuleContext, Context, parse_rulecontext};
+    /// let mut sig = Signature::default();
+    ///
+    /// let r =
+    ///     parse_rulecontext(&mut sig, "A(x_ [!]) = C(x_) | D([!])").expect("parse of A(x_ B[!]) = C(x_) | D([!])");
+    ///
+    /// let subcontexts: Vec<String> = r.subcontexts()
+    ///     .iter()
+    ///     .map(|(c, p)| format!("Subcontext:{},Place:{:?}", c.display(&sig), p))
+    ///     .collect();
+    ///
+    /// assert_eq!(
+    ///     subcontexts,
+    ///     vec![
+    ///         "Subcontext:A(x_ [!]),Place:[0]",
+    ///         "Subcontext:x_,Place:[0, 0]",
+    ///         "Subcontext:[!],Place:[0, 1]",
+    ///         "Subcontext:C(x_),Place:[1]",
+    ///         "Subcontext:x_,Place:[1, 0]",
+    ///         "Subcontext:D([!]),Place:[2]",
+    ///         "Subcontext:[!],Place:[2, 0]",
+    ///     ]
+    /// );
```

Comment:
Hmm...this doesn't seem very readable. Could you perhaps just use the following call to `format!`?

```rust
format!("({}, {:?})", c.display(&sig), p)
```